### PR TITLE
Added new section in the FAQ about # vs keyword, cleaned up styles

### PR DESCRIPTION
**Pull request by rickbutton**
I added a new section in the FAQ about `# vs keyword` pros/cons.

I also cleaned up a bit of the style. The doc is getting a bit long, so bumping up the section headers one size, and the subheaders one size, helps with readability IMO.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -326,15 +326,30 @@ const weakMap = new WeakMap();
...
+that match the "intent" of the feature, other than `const`. The `const` keyword is also tricky, because it describes
+a similar concept (variable reference immutability) while this proposal intends to add new immutable data structures.
+While immutability is the common thread between these two features, the features are orthogonal, and using `const`
+in both contexts would create hard-to-reason-about ambiguities.
 
-### Relation to const?
+Using the `#` character solves most of these problems. The `#` character can be safely used in this context without
+re-using an existing keyword or reserving a new keyword. Further, while `#` is already used to describe
+private fields in the [Class field declarations](https://github.com/tc39/proposal-class-fields) proposal, it is
+less ambiguous than re-using `const`, because they are completely orthogonal features (and contexts) with no relation
```
Comment (ljharb) [inline README.md]: This is a very subjective claim; i find it more ambiguous *because* they’re not related features.

**Comment by rricard** [comment]
Can you also propose the `{| |}`/`[| |]` alternative in that writeup as well as linking to a list of the reserved keywords.

**Comment by rickbutton** [comment]
@rricard can do!

File: README.md (hunk 1 above)
Comment (rricard) [reply README.md]: I agree, maybe saying that `const` had a strong community pushback would be more accurate

Comment (rickbutton) [reply README.md]: I agree, probably a little too much personal opinion, not enough fact. I've reworded it to better fit @rricard's suggestion of referencing the community pushback against `const` alone.

**Comment by rickbutton** [comment]
@rricard added the `{||}` and `[||]` blurb as requested, and added link to a reserved keyword list.

**Comment by rickbutton** [comment]
thanks @ljharb for the always super fast reviews!

**Pull request merged**